### ISSUE TITLE
No such file or directory @ sys_fail2

### ISSUE THREAD

**Issue by thejulian** [danmunn/redmine_dmsf#774]
_Redmine version 3.4.2.stable
Ruby version 2.1.5-p273 (2014-11-13) [x86_64-linux]
Rails version 4.2.8
redmine_dmsf 1.6.0_

Sometimes (it is completely random) file uploads are failing. 
I get following errors in the Interface:

`No such file or directory @ sys_fail2 - (/tmp/RackMultipart20170928-8186-1ni6zwq.zip, /srv/redmine/files/dmsf/2017/09/170928091910_4248_file.zip)`
"Some files were not commited due to validation errors: file.zip"

The interface shows the files, as they were uploaded correctly BUT they are not on the filesystem. So if you want to download them, you run in a 404 error. 

Currently I'm deleting the entries at the interface and reupload the same files. Most of the time the second try works. 

This is the production log:

```
Started POST "/projects/project/dmsf/upload/commit?folder_id=1631" for IP at 2017-09-28 09:19:10 +0200
Processing by DmsfUploadController#commit_files as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"", "commited_files"=>{"0"=>{"disk_filename"=>"170928091659_file.zip", "tempfile_path"=>"/tmp/RackMultipart20170928-8186-1ni6zwq.zip", "size"=>"207", "mime_type"=>"application/x-zip-compressed", "title"=>"title", "name"=>"file.zip", "description"=>"", "version"=>"1", "custom_version_major"=>"2", "custom_version_minor"=>"1", "comment"=>""}, "1"=>{"disk_filename"=>"170928091659_file.zip", "tempfile_path"=>"/tmp/RackMultipart20170928-8186-1im5z5h.zip", "size"=>"900441", "mime_type"=>"application/x-zip-compressed", "title"=>"title", "name"=>"title.zip", "description"=>"rel-2-3-25-1_r1271", "version"=>"1", "custom_version_major"=>"2", "custom_version_minor"=>"1", "comment"=>""}}, "name"=>"879 KB", "commit"=>"OK", "folder_id"=>"1631", "id"=>"project"}
  Current user: max.mustermann (id=13)
No such file or directory @ rb_sysopen - /tmp/RackMultipart20170928-8186-1ni6zwq.zip
No such file or directory @ sys_fail2 - (/tmp/RackMultipart20170928-8186-1ni6zwq.zip, /srv/redmine/files/dmsf/2017/09/170928091910_4248_file.zip)
No such file or directory @ rb_sysopen - /tmp/RackMultipart20170928-8186-1im5z5h.zip
No such file or directory @ sys_fail2 - (/tmp/RackMultipart20170928-8186-1im5z5h.zip, /srv/redmine/files/dmsf/2017/09/170928091910_4249_file.zip)
Redirected to https://redmine.com/projects/project/dmsf?folder_id=1631
Completed 302 Found in 234ms (ActiveRecord: 107.8ms)
```

**Comment by picman** [comment]
Fixed.

**Comment by thejulian** [comment]
Can you explain the root cause ? If you have time :-)

**Comment by picman** [comment]
Procedure of uploading files:
1. Rails creates a temporary file, e.g. `tmp/RackMultipart20170928-8186-1ni6cmd.sh`. 
2. DMSF creates an extra copy of this file, e.g. `tmp/171006092835_cmd.sh`.
3. A user makes a commit and DMSF moves this extra copy into a new document, e.g. `2017/10/171006092835_1356_cmd.sh`.

In order to speed up this process especially in case of big files I skipped step 2 and after committing I copied directly the temporary file created by Rails. Unfortunately sometimes the temporary file was deleted before the commit.
As a solution I rolled back to the original procedure with an extra copy of the file.
